src/Microsoft.AspNetCore.Razor.Evolution/DefaultDirectiveSyntaxTreePass.cs

```diff
@@ -0,0 +1,61 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System.Linq;
+using Microsoft.AspNetCore.Razor.Evolution.Legacy;
+
+namespace Microsoft.AspNetCore.Razor.Evolution
+{
+    internal class DefaultDirectiveSyntaxTreePass : IRazorSyntaxTreePass
```

Comment:
Chose to make this a syntax tree pass so by the time we get to generating IR errors are already logged for nested sections. Had some tests verifying that syntax tree parsing output the errors.